data/abi/abi_value.go

```diff
@@ -0,0 +1,304 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package abi
+
+import (
+	"fmt"
+	"math/big"
+)
+
+// Value struct is the ABI Value, holding ABI Type information and the ABI value representation.
+type Value struct {
+	ABIType Type
+	value   interface{}
+}
+
+// MakeUint8 takes a go `uint8` and gives an ABI Value of ABI type `uint8`.
+func MakeUint8(value uint8) (Value, error) {
+	bigInt := big.NewInt(int64(value))
+	return MakeUint(bigInt, 8)
+}
+
+// MakeUint16 takes a go `uint16` and gives an ABI Value of ABI type `uint16`.
+func MakeUint16(value uint16) (Value, error) {
+	bigInt := big.NewInt(int64(value))
+	return MakeUint(bigInt, 16)
+}
+
+// MakeUint32 takes a go `uint32` and gives an ABI Value of ABI type `uint32`.
+func MakeUint32(value uint32) (Value, error) {
+	bigInt := big.NewInt(int64(value))
+	return MakeUint(bigInt, 32)
+}
+
+// MakeUint64 takes a go `uint64` and gives an ABI Value of ABI type `uint64`.
+func MakeUint64(value uint64) (Value, error) {
```

Comment:
I think I would prefer these did not return errors.  They are constrained by the type system so that's impossible.  So it would be nice to not have to worry about the error returns in a public function.  IMO, it would be ok to panic on error internally.  Perhaps others disagree.

Reply:
I agree, for `MakeUint(8|16|32|64)`, the functions should probably just panic on error, since it should not be possible for an error to happen